src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1802 @@
+
+        # Linear Model for the onset of vaginal bleeding, in each 1 month period
+        # Create variables for used to predict the onset of vaginal bleeding at
+        # various stages of the disease
+
+        stage1 = p['r_vaginal_bleeding_cc_stage1']
+        stage2a = p['rr_vaginal_bleeding_cc_stage2a'] * p['r_vaginal_bleeding_cc_stage1']
+        stage2b = p['rr_vaginal_bleeding_cc_stage2b'] * p['r_vaginal_bleeding_cc_stage1']
+        stage3 = p['rr_vaginal_bleeding_cc_stage3'] * p['r_vaginal_bleeding_cc_stage1']
+        stage4 = p['rr_vaginal_bleeding_cc_stage4'] * p['r_vaginal_bleeding_cc_stage1']
+
+        self.lm_onset_vaginal_bleeding = LinearModel.multiplicative(
+            Predictor('sex').when('M', 0.0),
+            Predictor(
+                'ce_hpv_cc_status',
+                conditions_are_mutually_exclusive=True,
+                conditions_are_exhaustive=True,
+            )
+            .when('none', 0.00001)
+            .when('cin1', 0.00001)
+            .when('cin2', 0.00001)
+            .when('cin3', 0.00001)
+            .when('stage1', stage1)
+            .when('stage2a', stage2a)
+            .when('stage2b', stage2b)
+            .when('stage3', stage3)
+            .when('stage4', stage4)
+        )
+
+        # ----- DX TESTS -----
+        # Create the diagnostic test representing screening and the use of a biopsy
+        # in future could add different sensitivity according to target category
+
+        self.sim.modules['HealthSystem'].dx_manager.register_dx_test(
+            biopsy_for_cervical_cancer=DxTest(
+                property='ce_hpv_cc_status',
+                sensitivity=self.parameters['sensitivity_of_biopsy_for_cervical_cancer'],
+                target_categories=["stage1", "stage2a", "stage2b", "stage3", "stage4"]
+            )
+        )
+
+        self.sim.modules['HealthSystem'].dx_manager.register_dx_test(
+            screening_with_xpert_for_cin_and_cervical_cancer =DxTest(
+                property='ce_hpv_cc_status',
+                sensitivity=self.parameters['sensitivity_of_xpert_for_hpv_cin_cc'],
+                target_categories=["hpv", "cin1", "cin2", "cin3", "stage1", "stage2a", "stage2b", "stage3", "stage4"]
+            )
+        )
+
+        self.sim.modules['HealthSystem'].dx_manager.register_dx_test(
+            screening_with_via_for_cin_and_cervical_cancer=DxTest(
+                property='ce_hpv_cc_status',
+                sensitivity=self.parameters['sensitivity_of_via_for_cin_cc'],
+                target_categories=["cin1", "cin2", "cin3", "stage1", "stage2a", "stage2b", "stage3", "stage4"]
+            )
+        )
+
+        # ----- DISABILITY-WEIGHT -----
+        if "HealthBurden" in self.sim.modules:
+            # For those with cancer (any stage prior to stage 4) and never treated
+            self.daly_wts["stage_1_3"] = self.sim.modules["HealthBurden"].get_daly_weight(
```

Comment:
it might be mroe convenietn to have these map directly to the same categories you have defined in ce_hpv_cc_status